src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1311,15 +1312,19 @@ def iter_patches(
 
                 # Once we've dealt with any patches from the segment, update
                 # our position markers.
+                source_idx = segment.pos_marker.source_slice.stop
                 templated_idx = segment.pos_marker.templated_slice.stop
 
             # After the loop, we check whether there's a trailing deletion
             # or insert. Also valid if we still have an insertion buffer here.
             end_diff = self.pos_marker.templated_slice.stop - templated_idx
             if end_diff or insert_buff:
-                source_slice = segment.pos_marker.source_slice
+                source_slice = slice(
+                    source_idx,
+                    self.pos_marker.source_slice.stop,
+                )
                 templated_slice = slice(
-                    self.pos_marker.templated_slice.stop - end_diff,
+                    templated_idx,
```

Comment:
These are equivalent -- let's go back to high-school algebra. 🙂 

Above, we compute:
```
end_diff = self.pos_marker.templated_slice.stop - templated_idx
```

 So this:
```
self.pos_marker.templated_slice.stop - end_diff
```

is the same as this (replace `end_diff` with the formula):
```
self.pos_marker.templated_slice.stop - (self.pos_marker.templated_slice.stop - templated_idx)
```

Which is just:
```
templated_idx
```

Why did I bother with this?

1. It reads a little simpler.
2. It matches up with how we compute `source_slice` above without having to add a new `diff` variable, which would've made us want to rename `end_diff` as `templated_end_diff`, etc. etc.